File: bungee/src/main/java/com/emanuelvini/nuven/core/bungee/configuration/database/DatabaseValue.java
package com.emanuelvini.nuven.core.bungee.configuration.database;

import com.henryfabio.minecraft.configinjector.common.annotations.ConfigField;
import com.henryfabio.minecraft.configinjector.common.annotations.ConfigFile;
import com.henryfabio.minecraft.configinjector.common.injector.ConfigurationInjectable;
import lombok.Getter;
import lombok.experimental.Accessors;

import java.util.function.Function;

@Accessors(fluent = true)
@Getter
@ConfigFile("database.yml")
public class DatabaseValue implements ConfigurationInjectable {

    @Getter
    private static DatabaseValue instance = new DatabaseValue();

    @ConfigField("address")
    private String address;

    @ConfigField("username")
    private String username;

    @ConfigField("password")
    private String password;

    @ConfigField("database")
    private String database;

    public static <T> T get(Function<DatabaseValue, T> function) {
        return function.apply(instance);
    }

}

File: shared/src/main/kotlin/com/emanuelvini/nuven/core/shared/database/DatabaseValue.kt
package com.emanuelvini.nuven.core.shared.database

import com.henryfabio.minecraft.configinjector.common.annotations.ConfigField
import com.henryfabio.minecraft.configinjector.common.annotations.ConfigFile
import com.henryfabio.minecraft.configinjector.common.injector.ConfigurationInjectable
import java.util.function.Function


@ConfigFile("database.yml")
class DatabaseValue : ConfigurationInjectable {
    @ConfigField("address")
    val address: String? = null

    @ConfigField("username")
    val username: String? = null

    @ConfigField("password")
    val password: String? = null

    @ConfigField("database")
    val database: String? = null

    companion object {
        val instance = DatabaseValue()
        operator fun <T> get(function: Function<DatabaseValue, T?>): T? {
            return function.apply(instance)
        }
    }
}
